skcosmo/preprocessing/flexible_scaler.py

```diff
@@ -145,25 +132,34 @@ def fit(self, K=None, y=None, ref_cmean=None, ref_mean=None):
                     "The reference kernel is not square, and does not define a RKHS"
                 )
 
-            self.reference_shape = K.shape
+            self.reference_shape_ = K.shape
 
-            if ref_cmean is not None:
-                if K.shape[0] != len(ref_cmean):
+            if K_fit_rows is not None:
+                if K.shape[0] != len(K_fit_rows):
                     raise ValueError(
                         "The supplied column mean does not match the supplied kernel."
                     )
             else:
-                ref_cmean = K.mean(axis=0)
+                K_fit_rows = K.mean(axis=0)
 
-            if ref_mean is None:
-                ref_mean = K.mean()
+            if K_fit_all is None:
+                K_fit_all = K.mean()
 
         else:
-            assert ref_cmean is not None and ref_mean is not None
-            self.reference_shape = [None, len(ref_cmean)]
+            assert K_fit_rows is not None and K_fit_all is not None
+            self.reference_shape_ = [None, len(K_fit_rows)]
+
+        self.K_fit_rows_ = K_fit_rows
+        self.K_fit_all_ = K_fit_all
+
+        Kc = (
+            K
+            - np.broadcast_arrays(K, self.K_fit_rows_)[1]
+            - np.mean(K, axis=1).reshape((K.shape[0], 1))
+            + np.broadcast_arrays(K, self.K_fit_all_)[1]
+        )
 
-        self.ref_cmean = ref_cmean
-        self.ref_mean = ref_mean
+        self.scale_ = np.trace(Kc) / K.shape[0]
```

Comment:
Should we separate centering from the scaling so that we can do one without the other?